src/app/static/src/js/components/runWorkflow/runWorkflowReport.vue

```diff
@@ -53,34 +78,56 @@
 </template>
 
 <script lang="ts">
-import Vue from "vue"
-import {ReportWithDate, RunReportMetadata, RunWorkflowMetadata} from "../../utils/types";
+import Vue from "vue";
+import {BAlert} from "bootstrap-vue";
+import {
+    Parameter,
+    ReportWithDate,
+    RunReportMetadata,
+    RunWorkflowMetadata,
+    WorkflowReportWithParams
+} from "../../utils/types";
 import {api} from "../../utils/api";
 import GitUpdateReports from "../runReport/gitUpdateReports.vue";
+import ReportList from "../runReport/reportList.vue";
+import ParameterList from "../runReport/parameterList.vue";
 import ErrorInfo from "../errorInfo.vue";
+import {mapParameterArrayToRecord, mapRecordToParameterArray} from "../../utils/reports.ts";
+import {AxiosResponse} from "axios";
 
 interface Props {
-    workflowMetadata: RunWorkflowMetadata | null
+    workflowMetadata: RunWorkflowMetadata
 }
 
 interface Computed {
-    isReady: boolean
+    isReady: boolean,
+    hasReports: boolean,
+    reportParameters: Parameter[][],
+    stepIsValid: boolean
 }
 
 interface Methods {
-    getRunReportMetadata: () => void,
-    validateStep: () => void,
     branchSelected: (git_branch: string) => void,
     commitSelected: (git_commit: string) => void,
-    updateReports: (reports: ReportWithDate) =>  void
+    getParametersApiCall: (report: string) => Promise<AxiosResponse<any>>,
+    updateAvailableReports: (reports: ReportWithDate[]) =>  void,
```

Comment:
updateAvailableReport can be more descriptive, which available reports ? is it selectedReports or just a reportList ?

Reply:
It's the reports available for selection in the git commit - this changes when new commit selected in the git control, so requires this component to update the list available through the add report select. I've changed it to `updateAvailableReportsFromGit`, hope that's clearer.